ecommerce/views/v0/__init__.py

```diff
@@ -136,3 +169,500 @@ def create(self, request, *args, **kwargs):
+
+        internal_response = {
+            "state": "unknown or invalid",
+            "full_response": processor_response,
+        }
+
+        order = Order.objects.get(pk=converted_order.reference, purchaser=request.user)
+
+        if processor_response["decision"] == "DECLINE":
+            # Transaction declined for some reason
+            # This probably means the order needed to go through the process
+            # again so maybe tell the user to do a thing.
+            self.logger.debug(
+                "Transaction declined: {msg}".format(msg=processor_response["message"])
+            )
+        elif processor_response["decision"] == "ERROR":
+            # Error - something went wrong with the request
+            self.logger.debug(
+                "Error happened submitting the transaction: {msg}".format(
+                    msg=processor_response["message"]
+                )
+            )
+        elif processor_response["decision"] == "CANCEL":
+            # Transaction cancelled
+            # Transaction could be cancelled for reasons that don't necessarily
+            # mean that the entire order is invalid, so we'll do nothing with
+            # the order here.
+            self.logger.debug(
+                "Transaction cancelled: {msg}".format(msg=processor_response["message"])
+            )
+
+            internal_response = {
+                "state": "It was cancelled!",
+                "full_response": processor_response,
+                "order": converted_order,
+            }
+
+        elif processor_response["decision"] == "REVIEW":
+            # Transaction needs a professional review
+            # (nonjokey - someone needs to approve this so the order shouldn't be cancelled quite yet)
+            self.logger.debug(
+                "Transaction flagged for review: {msg}".format(
+                    msg=processor_response["message"]
+                )
+            )
+        elif processor_response["decision"] == "ACCEPT":
+            # It actually worked here
+            self.logger.debug(
+                "Transaction accepted!: {msg}".format(msg=processor_response["message"])
+            )
+
+            order.state = Order.STATE.FULFILLED
+            order.save()
+
+            internal_response = {
+                "state": "It worked!",
+                "full_response": processor_response,
+                "order": converted_order,
+            }
+        else:
+            # something else weird
+            self.logger.debug(
+                "Got this weird thing: {decision}\nMessage: {msg}".format(
+                    decision=processor_response["decision"],
+                    msg=processor_response["message"],
+                )
+            )
+
+        # for now let's just punt this back to the client
+        # cause i have no idea what to do yet
+
+        return Response(internal_response)
```

Comment:
I changed this to return just an empty response but really this needs to redirect to somewhere else in the app once it's finished processing the response (so this will all change as other things, notably https://github.com/mitodl/mitxonline/issues/387, get closer to completion).